javascript/openspatial-demo.js

```diff
@@ -0,0 +1,101 @@
+import fs from 'fs'
+
+import axios from 'axios'
+import FormData from 'form-data'
+import { waitForJob, getFirstMap } from './helpers.js'
+
+// Required constant values from the user
+const API_TOKEN = ''
+const USER_UUID = ''
+const ORGANIZATION_UUID = ''
+const PROJECT_UUID = ''
+const BASE_URL = ''
+
+
+const FILE_PATH = './Parks.zip'
+const TOOLS_URL = `${BASE_URL}/api/v2/tools/run?token=${API_TOKEN}`
+const JOBS_BASE_URL = `${BASE_URL}/api/v1/me/jobs`
+
+
+const TOOL_INPUT = {
+  clone_maps: true,
+  project_uuid: PROJECT_UUID,
+  user_uuid: USER_UUID
+}
+
+const openspatial_demo = async () => {
+  // 1. Check initial Area of Interest (Bounding box)
+
+  console.log('For initial map')
+  const initialMap = await getFirstMap(BASE_URL, PROJECT_UUID, API_TOKEN)
+
+  console.log('Area of Interest:', initialMap.boundingbox)
+  console.log('View:', initialMap.view)
+
+  // 2. Clone Project "OpenSpatial Demo"
+
+  const cloneResult = await axios.post(
+    // Endpoint
+    TOOLS_URL,
+    // Request body
+    {
+      tool_name: 'clone_project',
+      tool_input: TOOL_INPUT,
+      organization_uuid: ORGANIZATION_UUID
+    }
+  )
+
+  const cloneJobId = cloneResult.data.job_id
+
+  console.log('Cloning...')
+  const jobResult = await waitForJob(cloneJobId, JOBS_BASE_URL, API_TOKEN)
+
+  if (!Boolean(jobResult) || jobResult.status === 'FAILURE') throw 'Error while polling clone job'
```

Comment:
A failure doesn't indicate an error while polling, but that the job failed for some reason. In what case does the job result return a false? Any errors should be more through in order to demonstrate to the user the job's different statuses and what they mean.